eclipse-theia/theia
# Apply vscode Authentication plugin API

#### What it does

Apply vscode Authentication plugin API: https://github.com/microsoft/vscode/blob/ab42ffc44c709aded523b15399a070ae8c724824/src/vs/vscode.proposed.d.ts#L98-L131

Related CQ: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22526

#### How to test

1. Download the sample [plugin](https://github.com/vinokurig/authentication/releases/download/0.0.1/authentication-plugin-0.0.1.vsix) or clone and compile the sample project: https://github.com/vinokurig/authentication.git
2. Start the hosted plugin
3. Click the `Accounts` icon in the left bottom side and see:
![screenshot-localhost_3000-2020 08 31-11_55_24](https://user-images.githubusercontent.com/7668752/91702060-db8ccb00-eb80-11ea-90e0-6207ff541853.png)
4. Run `F1 => Authentication Sign in` command and click `Allow`:
![screenshot-localhost_3000-2020 08 31-11_57_08](https://user-images.githubusercontent.com/7668752/91702222-17c02b80-eb81-11ea-861a-7650f55fe2f0.png)
see the token message:
![screenshot-localhost_3030-2020 08 31-12_06_46](https://user-images.githubusercontent.com/7668752/91703218-8487f580-eb82-11ea-9071-07e3102c04d4.png)
5. Sign out from the account with the help of the `account` menu:
![screenshot-localhost_3030-2020 08 31-12_08_22](https://user-images.githubusercontent.com/7668752/91703379-c2851980-eb82-11ea-8655-7feaf482b147.png)
![screenshot-localhost_3030-2020 08 31-12_10_03](https://user-images.githubusercontent.com/7668752/91703474-ea747d00-eb82-11ea-8e6e-5c110ce7050f.png)
![screenshot-localhost_3000-2020 08 31-12_26_19](https://user-images.githubusercontent.com/7668752/91705094-2dcfeb00-eb85-11ea-85a6-2b83a471f1ac.png)
6. Click the `Accounts` icon and see the login item:
![screenshot-localhost_3000-2020 08 31-12_27_57](https://user-images.githubusercontent.com/7668752/91705256-6243a700-eb85-11ea-91a1-ff3f733b114b.png)
Click the login item and the token notification must appear:
![screenshot-localhost_3030-2020 08 31-12_06_46](https://user-images.githubusercontent.com/7668752/91707187-22ca8a00-eb88-11ea-9f0d-bd739b92a21e.png)
7. Run `F1 => Authentication Sign in` command and see:
![screenshot-localhost_3000-2020 08 31-12_52_31](https://user-images.githubusercontent.com/7668752/91707809-109d1b80-eb89-11ea-8333-cac5e34dce43.png)


vscode has `manage trusted extensions` feature that was not implemented because it requires quick-pick with multi-select: https://github.com/eclipse-theia/theia/issues/5673 
#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
